syft/pkg/cataloger/golang/licenses.go

```diff
@@ -68,38 +67,64 @@ func modCacheResolver(modCacheDir string) source.FileResolver {
 			log.Tracef("unable to open mod cache directory: %s, skipping mod cache resolver", modCacheDir)
 			r = source.NewMockResolverForPaths()
 		} else {
-			r = source.NewDeferredResolverFromSource(func() (source.Source, error) {
-				return source.NewFromDirectory(modCacheDir)
-			})
+			r = source.NewUnindexedDirectoryResolver(modCacheDir)
 		}
 	}
 
-	modCacheResolvers[modCacheDir] = r
-
 	return r
 }
 
 func (c *goLicenses) getLicenses(resolver source.FileResolver, moduleName, moduleVersion string) (licenses []string, err error) {
-	moduleName = processCaps(moduleName)
-
 	licenses, err = findLicenses(resolver,
-		fmt.Sprintf(`**/go/pkg/mod/%s@%s/*`, moduleName, moduleVersion),
+		fmt.Sprintf(`**/go/pkg/mod/%s@%s/*`, processCaps(moduleName), moduleVersion),
 	)
 
-	if c.searchLocalModCacheLicenses && err == nil && len(licenses) == 0 {
+	search := c.opts.searchLocalModCacheLicenses || c.opts.searchRemoteLicenses
```

Comment:
this seems to control the local search, but we're keying off of the remote search option too, is this intentional?

Reply:
this is intentional, because we're caching things in the same location